src/main/java/de/rub/nds/modifiablevariable/string/StringAppendValueModification.java

```diff
@@ -43,7 +46,12 @@ public void setAppendValue(final String appendValue) {
 
     @Override
     public VariableModification<String> getModifiedCopy() {
-        return new StringAppendValueModification(appendValue);
+        Random r = new Random();
+        int index = r.nextInt(appendValue.length());
```

Comment:
This makes no sense to me - first - we always try to be predictable - using a "true" random does not help in that. Second - I thought this is just a clone function - why would we even want to create a new modified string?

Reply:
Same issue seems to pop up in the string explicit value modidication

Reply:
and the string insert modification

Reply:
it feels more like you wanted to create  "createRandomModification" functions?

Reply:
(and the StringPrependModification)

Reply:
Soweit ich das in Erinnerung habe, habe ich einfach die Methoden korrigiert. Denn die waren einfach nicht korrekt für die `Modifications` von Strings implementiert. Bisher haben die einfach eine identische Modifikation geliefert. Aber wie der Name der Methode schon sagt "`getModifiedCopy()`" sollte denke ich eine modifizierte Version der Modifikation zurückgeliefert werden. Sonst wäre der Name eher "getCopy()"  

Siehe auch `getModifiedCopy()` von anderen `Modifications` anderer `ModifiableVariables` :

https://github.com/tls-attacker/ModifiableVariable/blob/75d4a316c51c97b11086091771b25160bd2d486b/src/main/java/de/rub/nds/modifiablevariable/integer/IntegerShiftLeftModification.java#L43C1-L57

https://github.com/tls-attacker/ModifiableVariable/blob/75d4a316c51c97b11086091771b25160bd2d486b/src/main/java/de/rub/nds/modifiablevariable/bytearray/ByteArrayExplicitValueModification.java#L59-L68

https://github.com/tls-attacker/ModifiableVariable/blob/75d4a316c51c97b11086091771b25160bd2d486b/src/main/java/de/rub/nds/modifiablevariable/biginteger/BigIntegerShiftRightModification.java#L49-L52

Auch dort wurde `java.util.Random` verwendet, eigentlich in jeder Modifikation. 

Interessanterweise wird in den Factories hingegen das `BadRandom` vom eigenen `RandomHelper` verwendet.
Siehe:
  https://github.com/tls-attacker/ModifiableVariable/blob/75d4a316c51c97b11086091771b25160bd2d486b/src/main/java/de/rub/nds/modifiablevariable/longint/LongModificationFactory.java#L89-L93


Also kurz gesagt, ich habe die String Modifikationen den anderen angepasst. Aber falls Änderungen daran gewünscht sind, kann ich das machen.

Reply:
:thinking: - tbh - those functions are not used anyways afaik - so it doesn't really matter anyways - if I get to refactoring the code they will get removed anyways so dw

Reply:
I know, that they are not used. I was already aware of this when I implemented the methods because I looked at how the various projects use modifiable variables. I also thought it was a bit funny. But I think that they could be useful.